tsdb/head.go

```diff
@@ -376,21 +373,82 @@ func (h *Head) loadWAL(r *wal.Reader, multiRef map[uint64]uint64) (err error) {
 			level.Warn(h.logger).Log("msg", "closing  memTombstones during wal read", "err", err)
 		}
 	}()
-	for r.Next() {
-		series, samples, tstones = series[:0], samples[:0], tstones[:0]
-		rec := r.Record()
-
-		switch dec.Type(rec) {
-		case record.Series:
-			series, err = dec.Series(rec, series)
-			if err != nil {
-				return &wal.CorruptionErr{
-					Err:     errors.Wrap(err, "decode series"),
+
+	var (
+		decoded    = make(chan interface{}, 10)
```

Comment:
Does increasing the channel size change the speed of loading WAL? (I don't know where the bottleneck is)

Reply:
I haven't played with channel size  much. Where the bottleneck is depends highly on shape of your data, hardware, and settings (such as WAL compression) so I think any value is going to be a guess. I chose a value pretty arbitrarily that gives a bit of headroom, but didn't make me concerned about increased memory usage...

Reply:
I tried with channel size 100 in my machine and no difference in the benchmark numbers. 10 sounds good then.